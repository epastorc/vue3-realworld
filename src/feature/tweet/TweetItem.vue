<template>
  <div class="tweet-container">
    <div class="tweet-avatar">
      <img :src="tweet.avatar" alt="avatar tweet" />
    </div>
    <div class="tweet-body">
      <div class="tweet-header">
        <span> <strong>{{ tweet.accountName }} </strong></span>
        <span class="no-principal-items"> {{ tweet.hashtag }} </span>
        <span class="no-principal-items"> {{ tweet.timeAgo }}</span>
      </div>
      <div class="tweet-content">
        <p>{{ tweet.content }}</p>
      </div>
    </div>
  </div>
</template>


<script lang="ts">
// eslint-disable-next-line no-unused-vars
import { defineComponent, PropType } from "vue";
// eslint-disable-next-line no-unused-vars
import { Tweet } from "./model/tweet";
export default defineComponent({
  name: "TweetItem",
  props: {
    tweet: {
      type:  Object as PropType<Tweet>,
    },
  },
})
</script>

<style scoped>
.no-principal-items{
    color: rgba(191, 191, 191, 1);
    margin-left: 5px;
}
.tweet-body {
  grid-column: 2 / 11;
}
.tweet-avatar {
  display: flex;
  justify-content: center;
  grid-column: 1 / 2;
}
.tweet-container {
  padding-top: 10px;
  display: grid;
  grid-template-columns: 1fr 9fr;
  border-bottom: 1px solid rgba(232, 236, 241, 1);
}
img {
  height: 40px;
  width: 40px;
  border-radius: 15px;
}
.tweet-header {
  display: flex;
}
.tweet-header > span {
  margin-right: 5px;
}
</style>