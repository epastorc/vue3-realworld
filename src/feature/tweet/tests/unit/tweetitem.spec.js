import { mount } from '@vue/test-utils'
import TweetItem from '../../TweetItem.vue'

describe('TweetItem.vue', () => {
  it('render component with props', () => {
      const tweet = {
        accountName: "test",
        avatar:"url",
        content:"content",
        timeAgo:"2ms",
        hashtag:"@test"
      }
    const wrapper = mount(TweetItem, {
        props: { tweet: tweet }
      })
    expect(wrapper).toMatchSnapshot();

  })
})
