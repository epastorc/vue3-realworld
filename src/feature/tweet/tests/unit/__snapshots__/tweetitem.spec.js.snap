// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetItem.vue render component with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {
        "0": Object {},
      },
      "__props": Object {
        "0": Array [
          Object {},
          Array [],
        ],
      },
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div
        class="tweet-container"
      >
        <div
          class="tweet-avatar"
        >
          <img
            alt="avatar tweet"
            src="url"
          />
        </div>
        <div
          class="tweet-body"
        >
          <div
            class="tweet-header"
          >
            <span>
              <strong>
                test
              </strong>
            </span>
            <span
              class="no-principal-items"
            >
              @test
            </span>
            <span
              class="no-principal-items"
            >
              2ms
            </span>
          </div>
          <div
            class="tweet-content"
          >
            <p>
              content
            </p>
          </div>
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "tweet": Object {
      "accountName": "test",
      "avatar": "url",
      "content": "content",
      "hashtag": "@test",
      "timeAgo": "2ms",
    },
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
